use denv v1.1.0's quiet no-overwrite init flags

tomeichlersmith opened the pull request:
This simplifies `just init` for newer `denv` users and avoids a confusing error message that is ignored because we immediately follow up a missing workspace with creating the workspace.

tomeichlersmith committed:
use denv v1.1.0's quiet no-overwrite init flags

this simplifies just init for newer denv and avoids a confusing error
message that is usually just passed by

tvami commented:
I tried to test this, but I get 
```
/run/user/3852/just/just-JaiBXy/init: line 107: [: : integer expression expected
```
on SLAC when I do `just init` in this branch

tvami commented:
Doing it on SLAC, where I had older `denv` I get
```
[tamasvami@sdfiana006 ldmx-sw]$ just init
/run/user/46072/just/just-GnFnRN/init: line 107: [: : integer expression expected
This directory does not exist. Would you like to create it for the denv? [Y/n] Y
mkdir: unrecognized option '--no-mkdir'
Try 'mkdir --help' for more information.
error: Recipe `init` failed with exit code 1
```
so maybe we should add another msg to make sure the user uses new enough `denv`

tomeichlersmith commented:
Apologies! I should have opened this as draft. I did not do any testing myself on anywhere outside my laptop. I'll mark as ready-for-review when I feel like it is ready.

tomeichlersmith committed:
patch up new init recipe

typo was preventing getting the minor version causing that check to
always fail. also putting quotes around all the environment variables
now as well.

tomeichlersmith committed:
have shell leave early if error

tomeichlersmith committed:
patch current and add new shellcheck recipe

new shellcheck recipe is focused on shellcheck'ing the recipes within
the justfile itself. As they get more complicated, it is helpful to make
sure that we maintain good shell style.

Don't leave shellchecks early on an error so that we can make sure that
the temporary files that are spawned are removed.

tomeichlersmith committed:
use xargs instead of cat

allows possibility of white-space characters in filepaths

tomeichlersmith commented:
@tvami @EinarElen are either of you available to double check this? I feel like its ready, but I don't want to merge it without double checks.

tvami commented:
I can have a look at this today!

tvami reviewed (approved):
I tested this, I have `denv v1.1.1` and it works nicely out of the box
